Fixes #35

jekirl opened the pull request:
Adds support for formatting instructions as per #35

paupino commented on tests/decimal_tests.rs:
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
+fn it_formats() {
+    let a = Decimal::from_str("233.323223").unwrap();
+    assert_eq!(format!("{}", a), "233.323223");
+    assert_eq!(format!("{:.9}", a), "233.323223");
```
Shouldn't this have 9dp padded? i.e. `233.323223000`?

paupino commented on tests/decimal_tests.rs:
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
+    let a = Decimal::from_str("233.323223").unwrap();
+    assert_eq!(format!("{}", a), "233.323223");
+    assert_eq!(format!("{:.9}", a), "233.323223");
+    assert_eq!(format!("{:.0}", a), "233.");
```
Should be `233` with no trailing `.`

paupino commented on tests/decimal_tests.rs:
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
+fn it_formats_neg() {
+    let a = Decimal::from_str("-233.323223").unwrap();
+    assert_eq!(format!("{}", a), "-233.323223");
+    assert_eq!(format!("{:.9}", a), "-233.323223");
```
As above, should be `-233.323223000`

paupino commented on tests/decimal_tests.rs:
```diff
@@ -85,6 +85,31 @@ fn it_can_serialize_deserialize() {
+    let a = Decimal::from_str("-233.323223").unwrap();
+    assert_eq!(format!("{}", a), "-233.323223");
+    assert_eq!(format!("{:.9}", a), "-233.323223");
+    assert_eq!(format!("{:.0}", a), "-233.");
```
As above, no trailing `.`

paupino commented on src/decimal.rs:
```diff
@@ -717,12 +721,7 @@ impl fmt::Display for Decimal {
-        }
-
-        f.pad(&rep)
+        f.pad_integral(self.is_positive(), "", &rep)
```
👍

jekirl commented:
Fixed the case where `precision > scale`. I had intentionally not add padded 0s since the behavior for `precision > MAX_PRECISION` could be a bit unexpected (as perceived increase in decimal precision).  Also added more tests and a fix for the corner case of ending up with an empty string after truncating.

paupino commented:
Looks good! I'm think your concern for increasing precision is warranted. I think perhaps in the future `error-chain` could be utilized and I replace some of the panic's with `Result`. The initial reason for `panic` doesn't hold true now with `?` and `map_err` etc.
